experiments/sharedUtils/BoxGenerator.js

```diff
@@ -1,127 +1,157 @@
+            }
+            dict[set] = box(setting)
+        }
+        return dict
+    }
     function box(setting){    
         let arr = [] //array of results
-        for(let i = 0; i < config.controls.length; i++){
-            arr.push(config.controls[i].eval(setting.args))
+        for(var i = 0; i < config.controls.length; i++){
+            arr.push(config.controls[i].eval(setting))
         }
         return arr //return the array
     }
-    return box //return the function
+    return configDict //return the function
 }
 
 // Takes in a configuration, and returns a string representation
 function toString(config){
     var dict = {}
-    for(let i = 0; i < config.controls.length; i++){
+    for(var i = 0; i < config.controls.length; i++){
         dict[`Light ${i+1}`] = {
-            type: config.controls[i].type
-        }
-        for(let j = 0; j < config.controls[i].arg.length; j++){
-            dict[`Light ${i+1}`][`b_${j+1}`] = config.controls[i].arg[j]
+            type: config.controls[i].type,
+            controls: config.controls[i].arg,
         }
     }
     return JSON.stringify(dict)
 }
 
-function createRandomBox(numButtons, numLights){
+function createRandomBox(numButtons, numLights, numStars){
     let configArray = []
     // create a control for each light
     for(let i = 0; i < numLights; i++){
-        console.log("creating a control for light: " + i)
-        let buttonArray = []
-        for(let j = 0; j < numButtons; j++){
-            buttonArray.push(Math.floor(Math.random()*2) === 1)
-        }
+        let compsArray = [] //array of components that control the light
+        let compsDict = {} //dict of components and their settings
+        do{
+            compsArray = []
+            compsDict = {}
+            for(let j = 0; j < numButtons; j++){
+                if(Math.floor(Math.random()*2) === 1){
+                    compsDict[`button${j+1}`] = (Math.floor(Math.random()*2) === 1)
+                    compsArray.push(`button${j+1}`)
+                }
+            }
+        } while(compsArray.length < 2 && numButtons >= 2); //keep going until there is at least one control
         let control
         switch(Math.floor(Math.random()*3)){
             case 0: //singular object
-                buttonArray.fill(false)
-                buttonArray[Math.floor(Math.random()*buttonArray.length)] = true
-                control = new SingObj(buttonArray)
+                let indexToChoose = Math.floor(Math.random()*numButtons)
+                var singCompsDict = {}
+                singCompsDict[`button${indexToChoose+1}`] = (Math.floor(Math.random()*2) === 1)
+                compsArray = [`button${indexToChoose+1}`]
+                control = new SingObj(singCompsDict, compsArray)
                 break
             case 1: //and object
-                control = new AndObj(buttonArray)
+                control = new AndObj(compsDict, compsArray)
```

Comment:
(see above) What happens when compsArray has only 1 value? Or is that not possible and I missed something